src/modelframe.jl

```diff
@@ -140,7 +140,9 @@ _droplevels!(x::Any) = x
 _droplevels!(x::AbstractCategoricalArray) = droplevels!(x)
 
 function ModelFrame(trms::Terms, d::AbstractDataFrame;
-                    contrasts::Dict = Dict())
+                    contrasts::Dict = Dict(), intercept::Bool = true)
+    trms = Terms((getfield(trms, fieldname) for fieldname in fieldnames(trms))...)
```

Comment:
Comment regarding the code rather than the content: `fieldnames` takes a type on 0.7, so we should make sure we have Compat with an appropriate lower bound as a dependency.